dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsDataS3ClientIT.java

```diff
@@ -56,137 +74,181 @@
-        List<MetricsData> metricsDataList = metricsDataClient.getMetricsData(toolId1, versionName);
-        assertEquals(1, metricsDataList.size());
-
-        // Verify the S3 folder structure
-        ListObjectsV2Request request = ListObjectsV2Request.builder().bucket(BUCKET_NAME).build();
-        ListObjectsV2Response listObjectsV2Response = s3Client.listObjectsV2(request);
-        List<S3Object> contents = listObjectsV2Response.contents();
-        assertEquals(1, contents.size());
-        assertEquals("workflow/github.com/ENCODE-DCC/pipeline-container%2Fencode-mapping-cwl/1.0/terra/" + fileName, contents.get(0).key());
-
-        // Verify that the S3 Object Metadata was recorded correctly
-        MetricsData metricsData = metricsDataList.get(0);
-        assertEquals(toolId1, metricsData.toolId());
-        assertEquals(versionName, metricsData.toolVersionName());
-        assertEquals(platform1, metricsData.platform());
-        assertEquals(fileName, metricsData.fileName());
-        assertEquals("workflow/github.com/ENCODE-DCC/pipeline-container%2Fencode-mapping-cwl/1.0/terra/" + fileName, metricsData.s3Key());
-
-        // Test getting the metadata of the metrics data
-        MetricsDataMetadata metricsDataMetadata = metricsDataClient.getMetricsDataMetadata(metricsData);
-        assertEquals(ownerUserId, metricsDataMetadata.owner());
-        assertEquals(description, metricsDataMetadata.description());
+    void testSubmitMetricsData() throws IOException {
+        // Admin user
+        final ApiClient webClient = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        final WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        final UsersApi usersApi = new UsersApi(webClient);
+        final ContainersApi containersApi = new ContainersApi(webClient);
+        final ExtendedGa4GhApi extendedGa4GhApi = new ExtendedGa4GhApi(webClient);
+        final String platform1 = Partner.TERRA.name();
+        final String platform2 = Partner.DNA_STACK.name();
+        final String description = "A single execution";
+        final Long ownerUserId = usersApi.getUser().getId();
 
-        // Verify S3 object contents
-        String metricsDataContent = metricsDataClient.getMetricsDataFileContent(metricsData.toolId(), metricsData.toolVersionName(), metricsData.platform(), metricsData.fileName());
-        assertEquals(metricsRequestBody, metricsDataContent);
+        // Register and publish a workflow
+        final String workflowId = "#workflow/github.com/DockstoreTestUser2/dockstore_workflow_cnv";
+        final String workflowVersionId = "master";
+        final String workflowExpectedS3KeyPrefixFormat = "workflow/github.com/DockstoreTestUser2/dockstore_workflow_cnv/master/%s"; // This is the prefix without the file name
+        Workflow workflow = workflowApi.manualRegister("github", "DockstoreTestUser2/dockstore_workflow_cnv", "/workflow/cnv.cwl", "", "cwl",
```

Comment:
there's an enum for "github" (SourceControl) and for descriptorlanguage "cwl" 
(helps when refactoring)